src/foam/dao/RoutingJournal.js

```diff
@@ -112,7 +122,9 @@ try {
               int length = line.trim().length();
               line = line.trim().substring(2, length - 1);
 
-              dao = (foam.dao.DAO) x.get(service);
+              // Touch the service to potentially trigger factory.
+              x.get(service);
```

Comment:
@daudich use of cmd_ might be appropriate again here.  Rather than have to do the magic of x.get() to trigger a factory that we know will update our map of dao names, we can just do dao = x.get().cmd_(GET_REPLAY_JOURNAL).

If we updated the original JDAO to respect this command we might not even have to write a new RoutingJDAO.